spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -1191,16 +1165,16 @@ public void run() {
 								}
 							}
 						}
-						if (SimpleMessageListenerContainer.this.idleEventInterval != null) {
+						if (getIdleEventInterval() != null) {
 							if (receivedOk) {
-								SimpleMessageListenerContainer.this.lastReceive = System.currentTimeMillis();
+								updateLastReceive();
 							}
 							else {
 								long now = System.currentTimeMillis();
 								long lastAlertAt = SimpleMessageListenerContainer.this.lastNoMessageAlert.get();
-								long lastReceive = SimpleMessageListenerContainer.this.lastReceive;
-								if (now > lastReceive + SimpleMessageListenerContainer.this.idleEventInterval
-										&& now > lastAlertAt + SimpleMessageListenerContainer.this.idleEventInterval
+								long lastReceive = getLastReceive();
+								if (now > lastReceive + getIdleEventInterval()
+										&& now > lastAlertAt + getIdleEventInterval()
```

Comment:
Why don't extract variable and don't call the method several times?
More over with the current `setIdleEventInterval(Long)` signature we can end up here with `NPE`...